README.md

```diff
@@ -1,6 +1,192 @@
+  .importFrom(new File("/home/user/Desktop/myPackage.jar"))
+  .as(JavaArchive.class);
+// or you can use more simple way for ZIP files
+JavaArchive archive = ShrinkWrap.createFromZipFile(JavaArchive.class, new File("/home/user/Desktop/myPackage.jar"));
+```
+
+### Domains
+
+In ShrinkWrap, a `Domain` encapsulates a configuration context for creating archives.
+Different domains can have isolated configurations, allowing you to manage multiple environments within a single application.
+
+
+#### Creating Domains
+
+You can create a new domain with a default configuration, a custom configuration, or a custom configuration builder.
+
+The `Configuration` class in ShrinkWrap encapsulates all the settings and properties for a given `Domain`.
+Each `Archive` created by the domain's `ArchiveFactory` will use this configuration internally.
+
+```java
+// Create a domain with default configuration
+Domain defaultDomain = ShrinkWrap.createDomain();
+
+// Create a domain with a custom configuration builder
+Domain domain = ShrinkWrap.createDomain(new ConfigurationBuilder().executorService(service)
+        .extensionLoader(loader));
+```
+
+#### Using Domains
+
+Once you have a domain, you can create archives within that domain, which will use previously specified configuration.
+
+```java
+// Create an archive within the default domain
+JavaArchive javaArchive = defaultDomain.getArchiveFactory().create(JavaArchive.class);
+
+// Create an archive within a custom domain
+JavaArchive customJavaArchive = customDomain.getArchiveFactory().create(JavaArchive.class, "customArchive.jar");
+
+// Add content to the archive
+customJavaArchive.addClass(MyClass.class);
+
+// Export the archive
+customJavaArchive.as(ZipExporter.class).exportTo(new File("/home/user/Desktop/customArchive.jar"), true);
+```
+
+### Filters
+
+ShrinkWrap provides filters to include or exclude specific content when creating archives.
+
+```java
+// Include all paths
+Filter<ArchivePath> includeAll = Filters.includeAll();
+
+// Include paths matching regex (e.g., all .class files)
+Filter<ArchivePath> includeClassFiles = Filters.include(".*\\.class");
+
+// Exclude paths matching regex (e.g., all .xml files)
+Filter<ArchivePath> excludeXmlFiles = Filters.exclude(".*\\.xml");
+
+// Include specific paths
+Filter<ArchivePath> includePaths = Filters.includePaths("path/to/include", "another/path");
+
+// Exclude specific paths
+Filter<ArchivePath> excludePaths = Filters.excludePaths("path/to/exclude", "another/path/to/exclude");
+
+// Apply filters to an archive
+JavaArchive archive = ShrinkWrap.create(JavaArchive.class, "myFilteredArchive.jar")
+    .addPackages(true, Filters.include("com.example.myapp.*"))  // Include specific package
+    .addAsResource(new FileAsset(new File("src/main/resources/config.xml")), "config.xml")  // Add specific resource
+    .filter(Filters.exclude(".*\\.log"));  // Exclude all .log files
+```
+
+### Issues and Improvements
```

Comment:
I would change the tone of this section. Instead of "Go here if you encounter issues" I'd use more "Go here if you want to contribute and fix issues". The section title would probably be "How to contribute" or something similar